### PR TITLE
2015 10 mn move filters and sort to list

### PR DESCRIPTION
**Pull request by MagdaN**
styling is missing

### CODE REVIEW

**Comment by MagdaN** [comment]
@local-girl : can you add the styling? Its in that task: https://tree.taiga.io/project/a3/task/711

**Comment by MagdaN** [comment]
looks good

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Listing.html [hunk 1]
```diff
@@ -1,4 +1,21 @@
 <div class="listing">
+    <div class="listing-action-bar">
+        <a ng-if="facets" class="listing-action-bar-link" data-ng-class="{ 'm-selected': showFilter }" data-ng-click="toggleFilter();" href="">Filter</a>
+        <a ng-if="sorts" class="listing-action-bar-link" data-ng-class="{ 'm-selected': showSort }" data-ng-click="toggleSort();" href="">Sort</a>
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Listing.html]: `data-` prefix

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Listing.ts [hunk 2]
```diff
@@ -69,13 +69,16 @@ export interface ListingScope<Container> extends angular.IScope {
     contentType? : string;
     facets? : IFacet[];
     sort? : string;
+    sorts?: string[];
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Listing.ts]: missing space before `:`

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Listing.html [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+    <div class="listing-results" ng-if="showFilter || showSort">
+        <adh-listfacets ng-if="showFilter" data-facets="facets" data-update="update" ></adh-listfacets>
+        <div ng-if="showSort">
+            <h3 class="sort-header">{{ "TR__SORT_BY" | translate }}</h3>
+            <a
+                href=""
+                class="button-cta-secondary sort-item"
+                data-ng-repeat="s in sorts"
+                data-ng-click="setSort(s)"
+                data-ng-class="{'is-active': sort === s }">{{ ('TR__' + s) | uppercase | translate }}
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Listing.html]: This way the translation keys can not be extracted automatically.

**Comment by xi** [comment]
I started reviewing but did not finish yet. Still to do:

-  fix translation keys (see https://github.com/liqd/adhocracy3.mercator/pull/1803#discussion_r44295270)
-  test on mercator
-  test that other listings/facets/sorts still work

Note that all other projects still need to be migrated. I agree that this should be done in a separate pull request though.

**Comment by xi** [comment]
It turns out that sort/filter are not used anywhere except for mercator. So we can remove the sidebar altogether (separate pull request).

One issue I found is that an empty `.listing-action-bar` still takes up some space, e.g. in the comment column. It will be changed in #1821 anyway but we should keep an eye on it.

I will merge this now and add a "pending review" tag because of my changes.

**Comment by MagdaN** [comment]
Looks good

**Pull request merged**